runelite-client/src/main/java/net/runelite/client/plugins/winemaking/WineMakingPlugin.java

```diff
@@ -0,0 +1,123 @@
+ * 2. Redistributions in binary form must reproduce the above copyright notice,
+ *    this list of conditions and the following disclaimer in the documentation
+ *    and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
+ * ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+ * WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR
+ * ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+ * ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+package net.runelite.client.plugins.winemaking;
+
+import com.google.common.eventbus.Subscribe;
+import com.google.inject.Provides;
+import java.awt.image.BufferedImage;
+import javax.inject.Inject;
+import net.runelite.api.Client;
+import net.runelite.api.InventoryID;
+import net.runelite.api.Item;
+import static net.runelite.api.ItemID.UNFERMENTED_WINE;
+import static net.runelite.api.ItemID.UNFERMENTED_WINE_1996;
+import static net.runelite.api.ItemID.ZAMORAKS_UNFERMENTED_WINE;
+import net.runelite.api.events.ConfigChanged;
+import net.runelite.api.events.GameTick;
+import net.runelite.api.queries.InventoryItemQuery;
+import net.runelite.client.config.ConfigManager;
+import net.runelite.client.game.ItemManager;
+import net.runelite.client.plugins.Plugin;
+import net.runelite.client.plugins.PluginDependency;
+import net.runelite.client.plugins.PluginDescriptor;
+import net.runelite.client.plugins.xptracker.XpTrackerPlugin;
+import net.runelite.client.ui.overlay.infobox.InfoBoxManager;
+
+@PluginDescriptor(
+	name = "Wine Making"
+)
+@PluginDependency(XpTrackerPlugin.class)
+public class WineMakingPlugin extends Plugin
+{
+	private static final InventoryItemQuery UNFERMENTED_WINE_QUERY;
+
+	static
+	{
+		UNFERMENTED_WINE_QUERY = new InventoryItemQuery(InventoryID.INVENTORY);
+		UNFERMENTED_WINE_QUERY.idEquals(UNFERMENTED_WINE, UNFERMENTED_WINE_1996, ZAMORAKS_UNFERMENTED_WINE);
+	}
+
+	@Inject
+	private Client client;
+
+	@Inject
+	private InfoBoxManager infoBoxManager;
+
+	@Inject
+	private WineMakingConfig config;
+
+	@Inject
+	private ItemManager itemManager;
+
+	private BufferedImage wineImage;
+
+	private int prevWineCount;
+
+	@Provides
+	WineMakingConfig getConfig(ConfigManager configManager)
+	{
+		return configManager.getConfig(WineMakingConfig.class);
+	}
+
+	@Override
+	protected void startUp() throws Exception
+	{
+		this.wineImage = itemManager.getImage(UNFERMENTED_WINE);
```

Comment:
This is probably unsafe and should be called from game thread

Reply:
This will crash when assertions are enabled. Please develop with `-ea` added to your JVM parameters.